### PR TITLE
Track how and when rules are deployed.

### PR DESCRIPTION
**Pull request by mbleigh**
Tracks a labeled analytics event for `inline` `json` and `bolt` rules.

### CODE REVIEW

File: lib/deploy/index.js [hunk 1]
```diff
@@ -79,6 +80,10 @@ var deploy = function(targetNames, options) {
       origin: api.uploadOrigin
     });
   }).then(function(res) {
+    if (_.get(options, 'config.notes.rules')) {
```
Comment (drtriumph) [inline lib/deploy/index.js]: `_.has` is more appropriate here I think

File: lib/config.js [hunk 2]
```diff
@@ -51,6 +52,7 @@ Config.prototype._materialize = function(target) {
   if (_.isString(this._src[target])) {
     return this._parseFile(target, this._src[target]);
   } else if (_.isObject(this._src[target])) {
+    if (target === 'rules') { this.notes.rules = 'inline'; }
```
Comment (drtriumph) [inline lib/config.js]: nit: can we put the block contents on a new line so we're consistent

**Comment by drtriumph** [comment]
Couple of nits, back to you

File: lib/deploy/index.js (hunk 1 above)
Comment (mbleigh) [inline lib/deploy/index.js]: I'm looking to see that it's set to something, not just that the key exists. If the code did `tihs.notes.rules = null` somewhere a `_.has` would still report it. I'd be willing to wrap in a `_.isString` if you'd prefer.

**Comment by drtriumph** [comment]
LGTM

**Pull request merged**